frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,372 @@
+local DataStorage = require("datastorage")
+local Event = require("ui/event")
+local FFIUtil = require("ffi/util")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local MultiConfirmBox = require("ui/widget/multiconfirmbox")
+local UIManager = require("ui/uimanager")
+local T = require("ffi/util").template
+local Utf8 = require("utf8")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local TimeVal = require("ui/timeval")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local _ = require("gettext")
+
+local ReaderUserHyph = WidgetContainer:new{
+    KEEP_ORIGINAL = 1, -- for dictionary file operations
+    USE_NEW = 2,       --     -"-
+}
+
+-----------------------------------------------
+--
+-----------------------------------------------
+
+-- returns path to the user dictionary
+function ReaderUserHyph:getDictionaryPath()
+    return FFIUtil.joinPath(DataStorage:getSettingsDir(),
+        self.ui.document:getTextMainLangDefaultHyphDictionary():gsub(".pattern$", "") .. "-user.hyph")
```

Comment:
I'd like to see a common prefix, `French-user.hyph` and `Spanish-user.hyph` would get far from each other among the rest of the files in some file explorer (although starting with uppercase, they may be at top in some file browsers).
`user-hyphenation-French.hyph` ?
`hyphenation-French-user.hyph` ?